xbmc/pvr/addons/PVRClient.cpp

```diff
@@ -1379,79 +1379,71 @@ std::shared_ptr<CPVRClientMenuHooks> CPVRClient::GetMenuHooks()
 PVR_ERROR CPVRClient::CallEpgTagMenuHook(const CPVRClientMenuHook& hook, const std::shared_ptr<CPVREpgInfoTag>& tag)
 {
   return DoAddonCall(__FUNCTION__, [&hook, &tag](const AddonInstance* addon) {
-    PVR_MENUHOOK_DATA hookData = {};
-    hookData.cat = PVR_MENUHOOK_EPG;
-    hookData.data.iEpgUid = tag->UniqueBroadcastID();
+    CAddonEpgTag addonTag(tag);
 
-    PVR_MENUHOOK menuHook = {0};
-    menuHook.category = hookData.cat;
+    PVR_MENUHOOK menuHook;
+    menuHook.category = PVR_MENUHOOK_EPG;
     menuHook.iHookId = hook.GetId();
     menuHook.iLocalizedStringId = hook.GetLabelId();
 
-    return addon->toAddon->MenuHook(menuHook, hookData);
+    return addon->toAddon->CallEPGMenuHook(&menuHook, &addonTag);
   });
 }
 
 PVR_ERROR CPVRClient::CallChannelMenuHook(const CPVRClientMenuHook& hook, const std::shared_ptr<CPVRChannel>& channel)
 {
   return DoAddonCall(__FUNCTION__, [&hook, &channel](const AddonInstance* addon) {
-    PVR_MENUHOOK_DATA hookData = {};
-    hookData.cat = PVR_MENUHOOK_CHANNEL;
-    WriteClientChannelInfo(channel, hookData.data.channel);
+    PVR_CHANNEL tag;
+    WriteClientChannelInfo(channel, tag);
 
-    PVR_MENUHOOK menuHook = {0};
-    menuHook.category = hookData.cat;
+    PVR_MENUHOOK menuHook;
+    menuHook.category = PVR_MENUHOOK_CHANNEL;
     menuHook.iHookId = hook.GetId();
     menuHook.iLocalizedStringId = hook.GetLabelId();
 
-    return addon->toAddon->MenuHook(menuHook, hookData);
+    return addon->toAddon->CallChannelMenuHook(&menuHook, &tag);
   });
 }
 
 PVR_ERROR CPVRClient::CallRecordingMenuHook(const CPVRClientMenuHook& hook, const std::shared_ptr<CPVRRecording>& recording, bool bDeleted)
 {
   return DoAddonCall(__FUNCTION__, [&hook, &recording, &bDeleted](const AddonInstance* addon) {
-    PVR_MENUHOOK_DATA hookData = {};
-    hookData.cat = bDeleted ? PVR_MENUHOOK_DELETED_RECORDING : PVR_MENUHOOK_RECORDING;
-    WriteClientRecordingInfo(*recording, hookData.data.recording);
+    PVR_RECORDING tag;
+    WriteClientRecordingInfo(*recording, tag);
 
-    PVR_MENUHOOK menuHook = {0};
-    menuHook.category = hookData.cat;
+    PVR_MENUHOOK menuHook;
+    menuHook.category = bDeleted ? PVR_MENUHOOK_DELETED_RECORDING : PVR_MENUHOOK_RECORDING;
     menuHook.iHookId = hook.GetId();
     menuHook.iLocalizedStringId = hook.GetLabelId();
 
-    return addon->toAddon->MenuHook(menuHook, hookData);
+    return addon->toAddon->CallRecordingMenuHook(&menuHook, &tag);
```

Comment:
Looks like here comes `menuHook.category` (see my previous comment) into play, but why this inconsistency? Either we introduce `CallRecordingMenuHook` and `CallDeletedRecordingMenuHook` and get rid of `PVR_MENUHOOK::category` or ... as I don't get the benefit of this whole change to make 5 (6?) `Call*MenuHook` functions out of the one API call we had before ... just revert this whole change?

Reply:
Have this related commit removed and becomes done on C++ rework.